File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        HashMap<Character, Integer> map = new HashMap<Character, Integer>();
        if(s.length() != t.length()) return false;
        for(int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            map.put(ch, map.getOrDefault(ch, 0) + 1);
        }
        for(int i = 0; i < t.length(); i++) {
            char ch = t.charAt(i);
            map.put(ch, map.getOrDefault(ch, 0) - 1);
        }
        
        for(char ch : map.keySet()) {
            if(map.get(ch) != 0) return false;
        }
        return true;
    }
}

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    
    let count = new Array(26);
    count.fill(0);
    for(let i = 0; i < s.length; i++) {
        count[s.charAt(i).charCodeAt() - 'a'.charCodeAt()]++;
    }
    for(let i = 0; i < t.length; i++) {
        count[t.charAt(i).charCodeAt() - 'a'.charCodeAt()]--;
    }
    for(let i= 0; i < 26; i++) {
        if(count[i] != 0) return false;
    }
    return true;
};